shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -87,3 +63,152 @@ async fn passthrough_standard() {
         10
     );
 }
+
+async fn test_create_index(client: &OpenSearch) {
+    client
+        .indices()
+        .create(IndicesCreateParts::Index("test-index"))
+        .send()
+        .await
+        .unwrap();
+
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&["test-index"]))
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(exists_response.status_code(), StatusCode::OK);
```

Comment:
If we assert that we get status ok, then the error reported to the developer is not very useful, all they know is that the status wasnt 200.
Instead, lets use a helper like this:
```rust
async fn assert_ok_and_get_json(response: Result<Response, Error>) -> Value {
    let response = response.unwrap();

    let status = response.status_code();
    let json = response.json::<Value>().await.unwrap();
    if status != StatusCode::OK {
        panic!("Opensearch query failed: {json:#?}");
    }
    json
}
```
Every query that we make to opensearch should use this helper, it will assert that the query succeeded, giving us the json if it did succeed.
If it fails then we get the error message from opensearch displayed in a panic.
e.g. I edited a query which was invalid, was I wrote this helper I got the following message which helped me understand what went wrong:
![Uploading image.png…]()

Reply:
Now that we know about `HEAD` requests, we should write this function like:
```rust
async fn assert_ok_and_get_json(response: Result<Response, Error>) -> Value {
    let response = response.unwrap();
    let status = response.status_code();

    if response.method() == Method::Head {
        if status != StatusCode::OK {
            panic!("Opensearch HEAD query returned status code {status}");
        }
        Value::Null
    } else {
        let json = response.json().await.unwrap();
        if status != StatusCode::OK {
            panic!("Opensearch query failed: {json:#?}");
        }
        json
    }
}
```